### DISCUSSION TITLE
dynamicTopoFvMesh/mesquitemotionsolver in blueCFD 2017-2

### DISCUSSION THREAD

**Discussion by ckr2611** [blueCFD/Core#239] [Ideas]
Hello, I was wondering if it is possible to compile and use the dynamicTopoFvMesh library of smenon in blueCFD 2017-2 ?

Thanks in advance

**Comment by wyldckat** [comment]
Sorry for taking a while to get back to you on this. I've thought about on and off, but most times that I looked into it for the past couple of years, I wasn't able to see any branches that would build with OpenFOAM, I would only see the ones for foam-extend.

Years ago, we did a port of it for an old blueCFD-Core version (maybe version 2.1?), but ended up not making it public due to lack of time to finish up the last touches on the thing... and I've regretted not having been able to finish it. Although at the time, Mesquite also had a weird license that made it harder to automate the build process, but I believe it's straight forward nowadays.

That said, there is a branch in dynamicTopoFvMesh for OpenFOAM 2.3.x: https://github.com/smenon/dynamicTopoFvMesh/tree/Port-2.3.x - although I suspect that not all features will work as intended, given Sandeep's report a few years ago: https://bugs.openfoam.org/view.php?id=1530 - and again, this might have been fixed since then...

So, in theory/principle, it should be possible, if it's already possible to build it with OpenFOAM 2.3.x. 
How easy it is, I'm not sure... but right now I don't have much time to look into this, given my priority task list for the next few months. That said, if you or your know someone willing to sponsor this work, I could allocate time at work to do this and shift priorities around a bit.

**Comment by ckr2611** [comment]
Ok, thank you for the information. I think will manage without it for the time being.

In any case, thank you for the awesome work in bringing OpenFOAM to Windows.
